### ISSUE TITLE
Crash when tray is enabled

### ISSUE THREAD

**Issue by TheSunCat** [JakeStanger/ironbar#166]
**Describe the bug**
If I enable the tray module, ironbar does not start.

**To Reproduce**
1. Enable the tray module
2. Try to launch ironbar

**Expected behavior**
ironbar should start

**System information:**
 - Distro: Arch Linux
- Compositor: Hyprland
 - Ironbar version: v0.12.0.r36.ge036ff0-1

**Configuration**
> Share your bar configuration and stylesheet as applicable:

<details><summary>Config</summary>

```
let {
    $tray = { type = "tray" }
    $end = [ $tray ]
}
in {
    position = "top"
    start = $start
    center = $center
    end = $end
}
```
</details>

**Additional context**
Crash log (with backtrace enabled):
[tray_crash.txt](https://github.com/JakeStanger/ironbar/files/11582443/tray_crash.txt)

**Comment by JakeStanger** [comment]
Interesting, I've not seen that before, but I've just tried it on my barebones Arch/Hyprland install and encountered the same thing. 

I'm not able to properly test at the moment, but I should be able to over the next couple of days. In the meantime it might be worth checking for any dependencies that are required for the service (I'm not aware of any other than dbus), or seeing if it occurs in different distros/environments. 

If the problem is replicable, it will likely be an issue with the [stray](https://github.com/oknozor/stray) library since that handles the dbus connection.

**Comment by TheSunCat** [comment]
Thanks for looking into this. I am not familiar with dbus but I've had issues before with Discord being unable to open links due to a missing dbus environment variable. I don't remember what it was as it is not an issue in WebCord, but perhaps ironbar needs this, too?

**Comment by JakeStanger** [comment]
I think that env var is possibly related to XDG, which would be separate to this. I've just updated Ironbar on my main machine and I'm now encountering the same thing, so it would seem something under the hood has changed.

**Comment by yavko** [comment]
I'm also experiencing this!!
![image](https://github.com/JakeStanger/ironbar/assets/15178513/a9ac97b6-b2d0-4189-9ad4-99cb6eb7fa52)

**Comment by JakeStanger** [comment]
I've managed to create a minimal repro, and it seems `stray` is crashing as soon as it tries to connect to DBus.

@TheSunCat are you using the stable Hyprland package or `-git`?

**Comment by TheSunCat** [comment]
I build Hyprland manually from git to add my own (unrelated) bugfixes.

**Comment by JakeStanger** [comment]
I've opened https://github.com/oknozor/stray/pull/7 with some fixes that seem to coax the tray back into life. 

I have just opened #193 here, which builds using the PR above. I'll merge that shortly. Because the fix relies on patching this crate, it will be necessary to install without using `cargo install`, even if the version there gets bumped.

Once the PR in stray is merged, I will be able to bump the package properly. I will leave this issue open until that's resolved.
